runtime/j9vm/java9vmi.c

```diff
@@ -951,93 +952,87 @@ JVM_DefineModule(JNIEnv * env, jobject module, jstring version, jstring location
+					success = (ERRCODE_SUCCESS == rc);
+					if (!success) {
+						throwExceptionHelper(currentThread, rc);
+					} else {
+						/* For "java.base" module setting of jrt URL and patch paths is already done during startup. Avoid doing it here. */
+						if (J9_ARE_ALL_BITS_SET(vm->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED)) {
+							Trc_MODULE_definition(currentThread, nameUTF);
+							if (classLoader == vm->systemClassLoader) {
+								success = vmFuncs->setBootLoaderModulePatchPaths(vm, j9mod, (const char *)nameUTF);
+								if (FALSE == success) {
+									vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+								} else {
+									const char* moduleName = "openj9.sharedclasses";
+	
+									if (0 == strcmp(nameUTF, moduleName)) {
+										J9VMDllLoadInfo *entry = FIND_DLL_TABLE_ENTRY(J9_SHARED_DLL_NAME);
+	
+										if ((NULL == entry)
+											|| (J9_ARE_ALL_BITS_SET(entry->loadFlags, FAILED_TO_LOAD))
+										) {
+											j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_VM_FAILED_TO_LOAD_MODULE_REQUIRED_DLL, J9_SHARED_DLL_NAME, moduleName);
+										}
 									}
 								}
 							}
+						} else {
+							/* first module; must be "java.base" */
+							J9ClassWalkState classWalkState;
+							J9Class* clazz = NULL;
+	
+							Assert_SC_true(0 == strcmp(nameUTF, JAVA_BASE_MODULE));
+	
+							clazz = vmFuncs->allClassesStartDo(&classWalkState, vm, vm->systemClassLoader);
+							while (NULL != clazz) {
+								Assert_SC_true(clazz->module == vm->javaBaseModule);
+								J9VMJAVALANGCLASS_SET_MODULE(currentThread, clazz->classObject, modObj);
+								clazz = vmFuncs->allClassesNextDo(&classWalkState);
+							}
+							vmFuncs->allClassesEndDo(&classWalkState);
+							vm->runtimeFlags |= J9_RUNTIME_JAVA_BASE_MODULE_CREATED;
+							TRIGGER_J9HOOK_JAVA_BASE_LOADED(vm->hookInterface, currentThread);
+							Trc_MODULE_definition(currentThread, "java.base");
 						}
-					} else {
-						/* first module; must be "java.base" */
-						J9ClassWalkState classWalkState;
-						J9Class* clazz = NULL;
-
-						Assert_SC_true(0 == strcmp(nameUTF, JAVA_BASE_MODULE));
-
-						clazz = vmFuncs->allClassesStartDo(&classWalkState, vm, vm->systemClassLoader);
-						while (NULL != clazz) {
-							Assert_SC_true(clazz->module == vm->javaBaseModule);
-							J9VMJAVALANGCLASS_SET_MODULE(currentThread, clazz->classObject, modObj);
-							clazz = vmFuncs->allClassesNextDo(&classWalkState);
-						}
-						vmFuncs->allClassesEndDo(&classWalkState);
-						vm->runtimeFlags |= J9_RUNTIME_JAVA_BASE_MODULE_CREATED;
-						TRIGGER_J9HOOK_JAVA_BASE_LOADED(vm->hookInterface, currentThread);
-						Trc_MODULE_definition(currentThread, "java.base");
-					}
-nativeOOM:
-					if (nameUTF != buf) {
-						j9mem_free_memory(nameUTF);
 					}
 					if (FALSE == success) {
 						vmFuncs->freeJ9Module(vm, j9mod);
```

Comment:
Should there be an assert here to validate that there is an exception pending?

Reply:
Added `Assert_SC_true`.